code/modules/mining/abandoned_crates.dm

```diff
@@ -0,0 +1,212 @@
+				new newcoin(src)
+		if(65 to 66)
+			new /obj/item/clothing/suit/ianshirt(src)
+		if(67 to 68)
+			var/t = rand(4,7)
+			for(var/i = 0, i < t, ++i)
+				var /newitem = pick(typesof(/obj/item/weapon/stock_parts) - /obj/item/weapon/stock_parts - /obj/item/weapon/stock_parts/subspace)
+				new newitem(src)
+		if(69 to 70)
+			for(var/i = 0, i < 5, ++i)
+				new /obj/item/bluespace_crystal(src)
+		if(71 to 72)
+			new /obj/item/weapon/pickaxe/drill(src)
+		if(73 to 74)
+			new /obj/item/weapon/pickaxe/drill/jackhammer(src)
+		if(75 to 76)
+			new /obj/item/weapon/pickaxe/diamond(src)
+		if(77 to 78)
+			new /obj/item/weapon/pickaxe/drill/diamonddrill(src)
+		if(79 to 80)
+			new /obj/item/weapon/cane(src)
+			new /obj/item/clothing/head/collectable/tophat(src)
+		if(81 to 82)
+			new /obj/item/weapon/gun/energy/plasmacutter(src)
+		if(83 to 84)
+			new /obj/item/toy/katana(src)
+		if(85 to 86)
+			new /obj/item/weapon/defibrillator/compact(src)
+		if(87) //1% chance
+			new /obj/item/weed_extract(src)
+		if(88)
+			new /obj/item/organ/brain(src)
+		if(89)
+			new /obj/item/organ/brain/alien(src)
+		if(90)
+			new /obj/item/organ/heart(src)
+		if(91)
+			new /obj/item/device/soulstone/anybody(src)
+		if(92)
+			new /obj/item/weapon/katana(src)
+		if(93)
+			new /obj/item/weapon/dnainjector/xraymut(src)
+		if(94)
+			new /obj/item/weapon/storage/backpack/clown(src)
+			new /obj/item/clothing/under/rank/clown(src)
+			new /obj/item/clothing/shoes/clown_shoes(src)
+			new /obj/item/device/pda/clown(src)
+			new /obj/item/clothing/mask/gas/clown_hat(src)
+			new /obj/item/weapon/bikehorn(src)
+			new /obj/item/toy/crayon/rainbow(src)
+			new /obj/item/weapon/reagent_containers/spray/waterflower(src)
+		if(95)
+			new /obj/item/clothing/under/rank/mime(src)
+			new /obj/item/clothing/shoes/sneakers/black(src)
+			new /obj/item/device/pda/mime(src)
+			new /obj/item/clothing/gloves/color/white(src)
+			new /obj/item/clothing/mask/gas/mime(src)
+			new /obj/item/clothing/head/beret(src)
+			new /obj/item/clothing/suit/suspenders(src)
+			new /obj/item/toy/crayon/mime(src)
+			new /obj/item/weapon/reagent_containers/food/drinks/bottle/bottleofnothing(src)
+		if(96)
+			new /obj/item/weapon/hand_tele(src)
+		if(97)
+			new /obj/item/clothing/mask/balaclava
+			new /obj/item/weapon/gun/projectile/automatic/pistol(src)
+			new /obj/item/ammo_box/magazine/m10mm(src)
+		if(98)
+			new /obj/item/weapon/katana/cursed(src)
+		if(99)
+			new /obj/item/weapon/storage/belt/champion(src)
+			new /obj/item/clothing/mask/luchador(src)
+		if(100)
+			new /obj/item/clothing/head/bearpelt(src)
+
+/obj/structure/closet/crate/secure/loot/attack_hand(mob/user as mob)
+	if(locked)
+		user << "<span class='notice'>The crate is locked with a Deca-code lock.</span>"
+		var/input = input(usr, "Enter [codelen] digits.", "Deca-Code Lock", "") as text
+		if(in_range(src, user) && !user.incapacitated())
```

Comment:
I removed the incapacitated because I assumed Miauw wanted me to do the whole line, if not then I'll put it back.